### DISCUSSION TITLE
So-suricata container missing post upgrade to 2.3.100

### DISCUSSION THREAD

**Discussion by bretthanus** [Security-Onion-Solutions/securityonion#7253] [Unsupported Versions]
cat /etc/soversion
2.3.100

On-prem deployment, installed on Centos7, from network, and soup'd from 2.3.61

Grid consists of 1 manager, 3 search nodes, 10 forward nodes (4 rebuilt to replace existing nodes attempting to diagnose this error i am reporting- normal forward node count would be 6)

no issues capturing network data, using span ports 
   
on manager, so-status shows all running(ok)

Grid shows all sensors at fault
on sensors, so-status shows everything running(ok) except suricata
so-status

`so-status

Checking Docker status

    Docker ------------------------------------------------------------ [ OK ]

Checking container statuses

    so-filebeat ------------------------------------------------------- [ OK ]
    so-nginx ---------------------------------------------------------- [ OK ]
    so-sensoroni ------------------------------------------------------ [ OK ]
    so-steno ---------------------------------------------------------- [ OK ]
    so-strelka-backend ------------------------------------------------ [ OK ]
    so-strelka-coordinator -------------------------------------------- [ OK ]
    so-strelka-filestream --------------------------------------------- [ OK ]
    so-strelka-frontend ----------------------------------------------- [ OK ]
    so-strelka-gatekeeper --------------------------------------------- [ OK ]
    so-strelka-manager ------------------------------------------------ [ OK ]
    so-suricata -------------------------------------------------- [ MISSING ]
    so-telegraf ------------------------------------------------------- [ OK ]
    so-wazuh ---------------------------------------------------------- [ OK ]
    so-zeek ----------------------------------------------------------- [ OK ]
`

salt-call state.highstate succeeds on manager node with zero failures:
	Summary for local
	--------------
	Succeeded: 750 (changed=23)
	Failed:      0
	--------------
	Total states run:     750
	Total run time:    96.408 s

Soup is successful:
`	SOUP - Security Onion UPdater

	Please review the following for more information about the update process and recent updates:
	https://docs.securityonion.net/soup
	https://blog.securityonion.net

	Press Enter to continue or Ctrl-C to cancel.

	### Preparing soup at Fri Feb 18 14:33:47 UTC 2022 ###

	Checking to see if this is a manager.

	This is a manager, so we can proceed.

	The manager's pillars can be rendered. We can proceed with SOUP.

	Checking to see if this is an airgap install.

	Found that Security Onion 2.3.100 is currently installed.

	Cloning Security Onion github repo into /tmp/sogh/securityonion.
	Removing previous upgrade sources.
	Cloning the Security Onion Repo.
	Cloning into 'securityonion'...
	remote: Enumerating objects: 53606, done.
	remote: Counting objects: 100% (3332/3332), done.
	remote: Compressing objects: 100% (1156/1156), done.
	remote: Total 53606 (delta 2195), reused 3108 (delta 2008), pack-reused 50274
	Receiving objects: 100% (53606/53606), 19.23 MiB | 29.59 MiB/s, done.
	Resolving deltas: 100% (34449/34449), done.
	Verifying we have the latest soup script.
	This version of the soup script is up to date. Proceeding.

	Elastic License has already been accepted

	Looks like you have an updated OS
	Generating new repo archive
	Let's see if we need to update Security Onion.
	Checking to see if there are hotfixes needed
	You are already running the latest version of Security Onion.

`

On every sensor, the minion log has this error:

		During handling of the above exception, another exception occurred:

		Traceback (most recent call last):
		  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 261, in render_tmpl
			output = render_str(tmplstr, context, tmplpath)
		  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 509, in render_jinja_tmpl
			raise SaltRenderError("Jinja variable {}{}".format(exc, out), buf=tmplstr)
		salt.exceptions.SaltRenderError: Jinja variable 'dict object' has no attribute 'gen_id'
		2022-02-18 14:53:48,056 [salt.state       :317 ][ERROR   ][98490] Unable to manage file: Jinja variable 'dict object' has no attribute 'gen_id'
		[root@at3infsosen01 salt]# tail minion -n20
		  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 502, in render_jinja_tmpl
			output = template.render(**decoded_context)
		  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
			self.environment.handle_exception()
		  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
			reraise(*rewrite_traceback_stack(source=source))
		  File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
			raise value.with_traceback(tb)
		  File "<template>", line 14, in top-level template code
		jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'gen_id'

		During handling of the above exception, another exception occurred:

		Traceback (most recent call last):
		  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 261, in render_tmpl
			output = render_str(tmplstr, context, tmplpath)
		  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 509, in render_jinja_tmpl
			raise SaltRenderError("Jinja variable {}{}".format(exc, out), buf=tmplstr)

salt-call state.highstate on sensor returns 2 errors (full log attached):
`	----------
			  ID: suriconfig
		Function: file.managed
			Name: /opt/so/conf/suricata/suricata.yaml
		  Result: True
		 Comment: File /opt/so/conf/suricata/suricata.yaml is in the correct state
		 Started: 15:17:02.484657
		Duration: 43.986 ms
		 Changes:
	----------
			  ID: surithresholding
		Function: file.managed
			Name: /opt/so/conf/suricata/threshold.conf
		  Result: False
		 Comment: Unable to manage file: Jinja variable 'dict object' has no attribute 'gen_id'
		 Started: 15:17:02.528882
		Duration: 40.501 ms
		 Changes:
	----------
			  ID: suribpf
		Function: file.managed
			Name: /opt/so/conf/suricata/bpf
		  Result: True
		 Comment: File /opt/so/conf/suricata/bpf is in the correct state
		 Started: 15:17:02.569638
		Duration: 4.063 ms
		 Changes:
	----------
			  ID: so-suricata
		Function: docker_container.running
		  Result: False
		 Comment: One or more requisite failed: suricata.surithresholding
		 Started: 15:17:02.583488
		Duration: 0.004 ms
		 Changes:`

my thresholding.conf.jinja file on master (/opt/so/saltstack/default/salt/suricata/files) has a datestamp that is within the start/end of serving soup on feb 4, hashing the file shows the file to be the same on manager node and sensor nodes
sha256sum threshold.conf.jinja
70db6cba3b9fe11c6bc4f48ef2be4703023656fc492440975294fb5dc84d71be  threshold.conf.jinja

Please let me know what else would help - thanks

brett
[master.gz](https://github.com/Security-Onion-Solutions/securityonion/files/8099315/master.gz)
[minion.gz](https://github.com/Security-Onion-Solutions/securityonion/files/8099316/minion.gz)
[soup.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/8099317/soup.log.gz)
[highstate.gz](https://github.com/Security-Onion-Solutions/securityonion/files/8099372/highstate.gz)

**Comment by dougburks** [comment]
> --- ID: surithresholding Function: file.managed Name: /opt/so/conf/suricata/threshold.conf Result: False Comment: Unable to manage file: Jinja variable 'dict object' has no attribute 'gen_id'

Double-check your thresholding configuration for syntax errors:
https://docs.securityonion.net/en/2.3/managing-alerts.html#threshold

**Reply by bretthanus** [reply]
Thanks- the first thing I considered was maybe i broke the 64 char length limit for IP field- but using `grep -n '^.\{64\}'` i checked all lines.  

I have checked the sensor nodes on which suricata container will not start, and the threshold.conf hasn't changed since November 2, but is the form of the pre-2.3 SO, "suppress gen_id 1, sig_id 2013926, track by_either, ip ..." 
IIRC the threshold.conf of this form had a limitation to use sig_id only once.  This threshold.conf seems to have been constructed using the global.sls yaml-ish form with the ip limitation of 64 characters, requiring to break long lists of IPs matching a sig_id:
`
suppress gen_id 1, sig_id 2017515, track by_src, ip 10.207.2.42,10.207.2.43,10.207.2.44,10.247.2.42,10.247.2.43
suppress gen_id 1, sig_id 2017515, track by_src, ip 10.207.2.108
suppress gen_id 1, sig_id 2017515, track by_src, ip 10.207.2.164
`
to simplify i changed the sensor local threshold.conf to contain the same single exception in each form- 
`
thresholding:
  sids:
    2013409:
      - suppress:
          gen_id: 1
          track: by_src
          ip: 10.207.1.146
`
and 
`
suppress gen_id 1, sig_id 2013409, track by_src, ip 10.207.1.146
`

setting threshold.conf to each of the above and trying so-start suricata still fails with 


[ERROR ] Rendering exception occurred
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 502, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 14, in top-level template code
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'gen_id'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 261, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 509, in render_jinja_tmpl
    raise SaltRenderError("Jinja variable {}{}".format(exc, out), buf=tmplstr)
salt.exceptions.SaltRenderError: Jinja variable 'dict object' has no attribute 'gen_id'
[ERROR   ] Unable to manage file: Jinja variable 'dict object' has no attribute 'gen_id'

and

`
----------
          ID: surithresholding
    Function: file.managed
        Name: /opt/so/conf/suricata/threshold.conf
      Result: False
     Comment: Unable to manage file: Jinja variable 'dict object' has no attribute 'gen_id'
     Started: 13:56:16.369478
    Duration: 927.581 ms
     Changes:
----------
          ID: suribpf
    Function: file.managed
        Name: /opt/so/conf/suricata/bpf
      Result: True
     Comment: File /opt/so/conf/suricata/bpf is in the correct state
     Started: 13:56:17.297571
    Duration: 9.045 ms
     Changes:
----------
          ID: so-suricata
    Function: docker_container.running
      Result: False
     Comment: One or more requisite failed: suricata.surithresholding
     Started: 13:56:17.468109
    Duration: 0.014 ms
     Changes:
`

**Reply by bretthanus** [reply]
this is probably going to be something self-inflicted.  I repeated the simplification exercise above on global.sls, to just one exception, and eventually the full grid came green. 
I am adding exceptions back to find out where I messed up- when I do find it - i'll update and close this in case it helps someone else- 
thanks!

**Reply by dougburks** [reply]
> I am adding exceptions back to find out where I messed up- when I do find it - i'll update and close this in case it helps someone else-

@bretthanus Were you able to fully resolve this?

**Comment by rweist** [comment]
Your problem is not self inflicted and the moderator telling someone to look at another solution instead of opening a new thread is wrong.  The pre-existing thread does not reflect the same things happening in my install/upgrade as what the others have reported.  I have no repository connection error.  I also don't have a "bond0" error.  I have a missing suricata on a brand new install.  I installed whatever was Pre 2.3.100 and it all worked.  I then went to 2.3.100 and suricata went missing. I upgraded to 2.3.110, and it is still missing.  I blew the entire array on my server away, re-installed the ISO, ran a distributed sensor install, and it still comes up missing.  This is a matter of not enough testing for failure on the part of Security Onion engineering.  Once, on 2.3.100, I re-ran the so-setup and it worked.  That lasted two days.  I can't consider this as an enterprise tool (even if I agree to pay support) if every time we upgrade, we might face a total redo of our entire environment.  The sad thing is that our upper level threat hunters love the interface and the data it gives them. It's better to them than other commercial products that we are currently demo`ing. BUT...failure after an upgrade is just not acceptable.

**Reply by dougburks** [reply]
@rweist Please review our Discussions guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720. If you would like us to help troubleshoot your issues, then please start a new discussion rather than replying to somebody else's discussion.
